metrics/resource_view_test.go

```diff
@@ -362,7 +363,14 @@ func TestMetricsExport(t *testing.T) {
 	}{{
 		name: "Prometheus",
 		init: func() error {
-			return UpdateExporter(context.Background(), configForBackend(prometheus), logtesting.TestLogger(t))
+			if err := UpdateExporter(context.Background(), configForBackend(prometheus), logtesting.TestLogger(t)); err != nil {
```

Comment:
just to ask the maybe-obvious question.. why is this function returning before the server is listening? Seems just about possible that this is not only a test bug if this can return before it's safe to ask for metrics so double-checking? (I guess it's not a big deal if metrics aren't immediately available since stuff generally polls?)

Reply:
Aside from stuff generally not really caring (as you said), I don't even know if there's a good way to make confidently sure it is serving without running health checks during the setup which seems excessive.

Reply:
I think generally you can split Listen and Serve if you want to be sure something is listening before returning. Saying that I totally buy that it probably isn't needed here, just instinctively feels a bit like the test may be be telling us the function is a bit fishy (clients need to poll before they can actually fetch metrics). I guess we can always fix the fishiness independently if we decide we care, and this solves the flake so lgtm.